### ISSUE TITLE
Add typing support

### ISSUE THREAD

**Issue by palfrey** [gruns/furl#148]
Currently running mypy against something using this project gets ```error: Skipping analyzing 'furl': found module but no type hints or library stubs```

**Comment by gruns** [comment]
do you have the time and bandwidth to add types to icecream? 🙂

**Comment by palfrey** [comment]
Turns out it's harder than that. `icecream` needs [executing](https://github.com/alexmojaki/executing), which needs [asttokens](https://github.com/gristlabs/asttokens) which needs [astroid](https://github.com/PyCQA/astroid). Astroid has had some typing work, but it still is a way off (see https://github.com/PyCQA/astroid/pull/1024#issuecomment-860230292). OTOH, if we ignore that, the others are more viable.

I've opened https://github.com/gristlabs/asttokens/pull/72 to start this process, but the build system appears a bit broken there.

**Comment by gruns** [comment]
@palfrey thank you for your efforts here! 🙌

**Comment by dsully** [comment]
Looks like the asttokens change has been merged.

**Comment by palfrey** [comment]
Yup. https://github.com/alexmojaki/executing/pull/57 is my initial work there

**Comment by palfrey** [comment]
And icecream with https://github.com/gruns/icecream/pull/167

**Comment by palfrey** [comment]
I was going to finally look at this, but I'm blocked on https://github.com/gruns/furl/issues/185 and https://github.com/gruns/furl/issues/164

**Comment by gruns** [comment]
hey hey @palfrey!

follow-ups always helpful! unfortunately i dont have time to tackle these right now 

@alexcochran just joined as a maintainer. 🙌 tagging him. perhaps he has time to lean in and avail here 🤞
